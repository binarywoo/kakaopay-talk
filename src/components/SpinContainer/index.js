import React from 'react'
import { Spin } from 'antd'

const SpinContainer = ({ height = '47vh', size, tip }) => {
  return (
    <div
      style={{ textAlign: 'center', paddingTop: height, paddingBottom: height }}
    >
      <Spin size={size} tip={tip} />
    </div>
  )
}

export default SpinContainer
