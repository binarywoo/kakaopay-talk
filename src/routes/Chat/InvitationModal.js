import React from 'react'
import { Modal, List } from 'antd'
import moment from 'moment'
import SpinContainer from '../../components/SpinContainer/index'

const InvitationModal = ({
  show,
  userList,
  chat,
  user,
  onClose,
  onClickUser
}) => {
  return (
    <Modal
      visible={show}
      title='대화상대 초대'
      footer={null}
      onCancel={onClose}
    >
      {!userList && <SpinContainer height='30vh' />}
      {userList && (
        <List
          style={{
            padding: '0 12px'
          }}
          itemLayout='horizontal'
          dataSource={userList.filter(
            item => item.key !== user.key && !chat.users[item.key]
          )}
          renderItem={user => {
            moment.locale('ko')
            return (
              <List.Item
                onClick={() => onClickUser(user)}
                style={{ cursor: 'pointer' }}
              >
                <List.Item.Meta title={user.userId} />
              </List.Item>
            )
          }}
        />
      )}
    </Modal>
  )
}

export default InvitationModal
