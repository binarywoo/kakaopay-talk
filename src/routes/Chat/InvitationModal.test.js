import React from 'react'
import { shallow, mount } from 'enzyme'
import InvitationModal from './InvitationModal'
import SpinContainer from '../../components/SpinContainer/index'
import { List } from 'antd'

describe('InvitationModal', () => {
  let wrapper = null

  const props = {
    show: true,
    userList: ['a', 'b', 'c', 'd', 'e', 'f'].map(item => ({
      key: item,
      userId: item
    })),
    chat: {
      users: {
        a: true,
        b: true,
        c: true
      }
    },
    user: { key: 'a', userId: 'a' },
    onClose: () => null,
    onClickUser: () => null
  }

  it('정상적으로 렌더링이 된다.', () => {
    wrapper = shallow(<InvitationModal {...props} />)
  })

  it('스냅샷과 매칭이 된다.', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('유저 목록이 로딩 되었을 때 자신을 제외하고, 현재 방에 없는 유저 List를 보여준다.', () => {
    wrapper = mount(<InvitationModal {...props} />)
    expect(wrapper.find(List).length).toBe(1)
    expect(wrapper.find(List.Item).length).toBe(3)
  })

  it('유저 목록이 아직 로딩되지 않았을 때 SpinContainer를 보여준다.', () => {
    const newProps = Object.assign({}, props, {
      userList: null
    })
    wrapper = shallow(<InvitationModal {...newProps} />)
    expect(wrapper.find(SpinContainer).length).toBe(1)
  })
})
