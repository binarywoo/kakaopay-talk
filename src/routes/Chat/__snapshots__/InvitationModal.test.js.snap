// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitationModal 스냅샷과 매칭이 된다. 1`] = `
<Modal
  cancelButtonDisabled={false}
  confirmLoading={false}
  footer={null}
  maskTransitionName="fade"
  okButtonDisabled={false}
  okType="primary"
  onCancel={[Function]}
  title="대화상대 초대"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "key": "d",
          "userId": "d",
        },
        Object {
          "key": "e",
          "userId": "e",
        },
        Object {
          "key": "f",
          "userId": "f",
        },
      ]
    }
    itemLayout="horizontal"
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
    style={
      Object {
        "padding": "0 12px",
      }
    }
  />
</Modal>
`;
